protocol-units/da-sequencer/node/src/lib.rs

```diff
@@ -113,13 +137,31 @@ where
 	}
 }
 
+async fn stream_block_to_sender(
+	senders: &mut Vec<mpsc::UnboundedSender<Option<SequencerBlock>>>,
+	data: Option<SequencerBlock>,
+) {
+	let mut new_sender = vec![];
+	for sender in senders.drain(..) {
+		// Remove the sender in error because it means the client was disconnected.
+		if let Err(err) = sender.send(data.clone()) {
+			tracing::warn!("Failed to send block to grpc client. Client disconnected. remove connection :{err}");
+		} else {
+			new_sender.push(sender);
+		}
+	}
+	*senders = new_sender;
+}
+
 /// manage the optional future for block production.
 async fn conditional_block_producing(
-	fut: Option<&mut JoinHandle<Result<Option<SequencerBlock>, DaSequencerError>>>,
+	opt_fut: &mut Option<JoinHandle<Result<Option<SequencerBlock>, DaSequencerError>>>,
```

Comment:
Function parameters shouldn't be of type `Option` unless they provide some additional (optional) context. In this case, if `opt_fut` is `None`, don't call the function.

Reply:
Yeah I agree with @sebtomba here, I thought though it was some special case here, let's stick to `fut`.

Reply:
If you have a better solution for this use case, I take it.

Reply:
In fact, the future is not cancel safe, so I've to change this. I give a new try.

Reply:
I've a new proposition